modules/storages/app/common/storages/adapters/providers/sharepoint/services/create_managed_folders_service.rb

```diff
@@ -0,0 +1,175 @@
+
+            class << self
+              def call(storage:, project_storages_scope: nil)
+                new(storage:, project_storages_scope:).call
+              end
+            end
+
+            def initialize(storage:, project_storages_scope:, hide_missing_folders: project_storages_scope.nil?)
+              super()
+              @storage = storage
+              @hide_missing_folders = hide_missing_folders
+              @project_storages = (project_storages_scope || @storage.project_storages).active.automatic
+            end
+
+            def call
+              with_tagged_logger([self.class.name, "storage-#{@storage.id}"]) do
+                remote_folders_map.bind do |existing_remote_folders|
+                  ensure_folders_exist(existing_remote_folders).bind do
+                    hide_inactive_folders(existing_remote_folders) if @hide_missing_folders
+                  end
+                end
+
+                @result
+              end
+            end
+
+            private
+
+            def ensure_folders_exist(folder_map)
+              info "Ensuring that automatically managed project folders exist and are correctly named."
+              @project_storages.includes(:project).find_each do |project_storage|
+                unless folder_map.key?(project_storage.project_folder_id)
+                  info "#{project_storage.managed_project_folder_path} does not exist. Creating..."
+                  next create_remote_folder(project_storage.managed_project_folder_name, project_storage)
+                end
+
+                rename_project_folder(folder_map[project_storage.project_folder_id], project_storage)
+              end
+
+              Success(:folder_maintenance_done)
+            end
+
+            def hide_inactive_folders(folder_map)
+              info "Hiding folders related to inactive projects"
+
+              inactive_folder_ids(folder_map).each { |item_id| hide_folder(item_id, folder_map) }
+            end
+
+            def hide_folder(file_id, folder_map)
+              info "Hiding folder with ID #{file_id} as it does not belong to any active project"
+
+              input_data = Adapters::Input::SetPermissions.build(file_id:, user_permissions: []).value_or do |failure|
+                log_validation_error(failure, file_id:, context: "hide_folder")
+              end
+
+              set_permissions.call(auth_strategy:, input_data:).or do |error|
+                add_error(:hide_inactive_folders, error, options: { context: "hide folders", path: folder_map[item_id] })
+              end
+            end
+
+            def inactive_folder_ids(folder_map)
+              folder_map.keys - @project_storages.pluck(:project_folder_id).compact
+            end
+
+            def rename_project_folder(current_folder_name, project_storage)
+              actual_name = project_storage.managed_project_folder_name
+              return if current_folder_name == actual_name
```

Comment:
:yellow_circle: It's confusing to me that the original implementation compares the `managed_project_folder_path` to a variable called `current_folder_name`. So your implementation is more consistent, because both are called `*_name`.

That said, I am not sure what the `folder_map` that feeds this contains. Names or pathes. I am wondering whether our test suite is dense enough to spot an error w.r.t. mixing those two up (I guess special characters are needed to do so).